homeassistant/components/generic/config_flow.py

```diff
@@ -314,19 +343,46 @@ async def async_step_user(
                         # The automatically generated still image that stream generates
                         # is always jpeg
                         user_input[CONF_CONTENT_TYPE] = "image/jpeg"
-
-                    return self.async_create_entry(
-                        title=name, data={}, options=user_input
+                    self.user_input = user_input
+                    self.title = name
+
+                    # Register a temporary view so that we can show a preview
+                    register_preview(hass, self.flow_id, user_input)
+                    preview_url = f"/api/generic/preview_flow_image/{self.flow_id}?t={datetime.now().isoformat()}"
+                    return self.async_show_form(
```

Comment:
Don't move directly to another step's form. Call the step method.

Showing a form is a private action of the step method.

Reply:
Ok understood.  Should be sorted now in 02b94940